### ISSUE TITLE
Bug on created meshgrid of the NetCDF file

### ISSUE THREAD

**Issue by kasra-keshavarz** [ShervanGharari/candex#1]
Hello,

I've created a meshgrid of the netcdf file that I have and it seems that the created meshgrid does not cover the area covered by the netCDF file.

This is the code that I've used:

```python
from candex import *

import netCDF4 as nc

ncfile = nc.Dataset("sample_netcdf_era5.nc")
lats = ncfile.variables['latitude'][:]
lons = ncfile.variables['longitude'][:]
result = NetCDF_SHP_lat_lon('forcing_1979.nc', [48, 56, -119, -99], 'latitude', 'longitude', False)

result.to_file('era5.shp')
```

I've attached a sample netCDF file that I've used : 
[sample_netcdf_era5.zip](https://github.com/ShervanGharari/candex/files/3611988/sample_netcdf_era5.zip)

Here is a screenshot:
![CandexBUG](https://user-images.githubusercontent.com/50383939/64898153-42993300-d643-11e9-93bc-047eef522427.JPG)

**Comment by ShervanGharari** [comment]
Hi Kasra,

Thank you for posting this issue. But actually this is done intentionally. There are two ways to create the shapefile for a grids which are on the edge.

1. Approximate the grid cell size; this can easily be done if the grids are spaces with a fixed increment of lat/lon (for example 0.25 degree as it is the case for ERA-5). This approximation get a bit tricky when dealing with rotated lat/lon as each grid has different lat/lon increments from its neighboring grids.

2. Simply ignore the grids on edge; this is what candex does. The assumption here is that the existing netcdf file is subset assuming a buffer around the actual target shapefile (a basin for example).

in candex, when a shapefile is provided, such as a basin, the ```box``` function considers a buffer to create the min/max of lat/lon for creating shapefile for Netcdf grids. However, if you give this box values of max/min lat/lon manually, as you did, you should consider this buffer also manually. In your case it will be sth like:

```
result = NetCDF_SHP_lat_lon('forcing_1979.nc', [47.75, 56.25, -119.25, -98.75], 'latitude', 'longitude', False)
```

Hope this answers your concern.

**Comment by kasra-keshavarz** [comment]
Thanks for the comments. I guess the issue is solved.
